treesit-auto.el

```diff
@@ -335,28 +305,33 @@ Non-nil only if installation completed without any errors."
     ;; work in the future.
     (not (treesit-install-language-grammar lang))))
 
-(defun treesit-auto--maybe-install-grammar ()
-  "Try to install the grammar matching the current major-mode.
-
-If the tree-sitter grammar is missing for the current major mode,
-this will silently fail, automatically install the grammar, or
-prompt the user about automatic installation, depending on the
-value of `treesit-auto-install'.  If installation of the grammar
-is successful, activate the tree-sitter major mode."
-  (when-let* ((not-ready (not (treesit-auto--ready-p major-mode)))
-              (lang (alist-get major-mode treesit-auto-mode-lang-alist))
-              (recipe (alist-get lang treesit-auto-lang-recipe-alist))
-              (ts-mode (treesit-auto-recipe-ts-mode recipe))
-              (ts-mode-exists (fboundp ts-mode))
-              (install-success (treesit-auto--prompt-to-install-package lang)))
-    (funcall ts-mode)))
-
 (defvar treesit-auto-opt-out-list
   nil
   "Language symbols to avoid when using `treesit-auto-install-all'.
 
 This variable is ignored if `treesit-auto-langs' is non-nil.")
 
+(defun treesit-auto--get-mode-recipe (&optional mode)
+  (let ((mode (or mode major-mode)))
+    (cl-loop for recipe in treesit-auto-recipe-list
+             if (or (equal (treesit-auto-recipe-remap recipe) mode)
```

Comment:
Issue: as already mentioned, this probably requires `ensure-list` along with `memq`, in to handle either a list *or* a single symbol provided to `:remap` in the recipe

Reply:
ah, did not realise that remap can be a list. will change that.